client/my-sites/email/titan-new-mailbox-list/index.jsx

```diff
@@ -75,29 +80,45 @@ const TitanNewMailboxList = ( {
 	return (
 		<div className="titan-new-mailbox-list__main">
 			{ mailboxes.map( ( mailbox, index ) => (
-				<TitanNewMailbox
-					key={ mailbox.uuid }
-					onMailboxAdd={ onMailboxAdd }
-					onMailboxRemove={ onMailboxRemove( mailboxes, mailbox.uuid ) }
-					onMailboxValueChange={ onMailboxValueChange( mailbox.uuid ) }
-					mailbox={ mailbox }
-					onReturnKeyPress={ onReturnKeyPress }
-					showAllErrors={ validatedMailboxUuids.includes( mailbox.uuid ) }
-					showLabels={ showLabels }
-					showTrashButton={ index > 0 }
-				/>
+				<React.Fragment key={ `${ index }:${ mailbox.uuid }` }>
+					{ index > 0 && (
+						<CardHeading
+							className="titan-new-mailbox-list__numbered-heading"
+							tagName="h3"
+							size={ 20 }
+						>
+							Mailbox { String( index + 1 ).padStart( 2, '0' ) }
+						</CardHeading>
+					) }
+
+					<TitanNewMailbox
+						onMailboxValueChange={ onMailboxValueChange( mailbox.uuid ) }
+						mailbox={ mailbox }
+						onReturnKeyPress={ onReturnKeyPress }
+						showAllErrors={ validatedMailboxUuids.includes( mailbox.uuid ) }
+						showLabels={ showLabels }
+					/>
+
+					<div className="titan-new-mailbox-list__actions">
+						{ showAddAnotherMailboxButton && (
+							<Button onClick={ () => onMailboxAdd( index ) }>
+								<Gridicon icon="plus" />
+								<span>{ translate( 'Add another mailbox' ) }</span>
+							</Button>
+						) }
+						{ index > 0 && (
+							<Button onClick={ onMailboxRemove( mailboxes, mailbox.uuid ) }>
+								<Gridicon icon="trash" />
+								<span>{ translate( 'Remove this mailbox' ) }</span>
```

Comment:
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **18** times:
`translate( 'Remove mailbox' )` *ES Score: 6*